src/arithmetic/modular_set.cc

```diff
@@ -36,6 +36,18 @@ struct ModularSetAnalyzer::Entry {
   int64_t coeff{1};
   int64_t base{0};
 
+  Entry() = default;
+
+  Entry(int64_t coeff, int64_t base) {
+    CHECK_GE(coeff, 0);
```

Comment:
For one case in `test_arith_rewrite_simplify.py`, `coeff` can be smaller than 0

Reply:
Because this constructor is too "smart", we can create a static function.
see https://github.com/dmlc/tvm/pull/2726#discussion_r267017925

Reply:
In my opinion, ```coeff``` should not be smaller than 0 (see https://github.com/dmlc/tvm/pull/2726#discussion_r267017300). I have no idea about the test cases in ```test_arith_rewrite_simplify.py```  but I do pass these tests. Further more, if this case  exists in unit test, probably we can try to fix it.